api/include/opentelemetry/trace/trace_state.h

```diff
@@ -107,11 +107,38 @@ class TraceState
     }
   };
 
+  static TraceState &GetDefault()
+  {
+    static TraceState ts;
+    return ts;
```

Comment:
I think this doesn't give us too much benefit here, considering that the `TraceState` object is copied on assignment anyway. This pattern might make more sense when using a shared pointer:
```c++
static nostd::shared_ptr<TraceState> ts(new TraceState());
return ts;
```

Reply:
Good point. Passing by reference, or shared ptr would have benefit in this case. And if we are planning to store trace_state as `shared_ptr`  in span context, returning as shared_ptr would be better option.

Reply:
As the return `TraceState` is mutable, I'd be hesitate to make it as static. Changed it to create a new empty instance and renamed it as `GetEmpty()`.

Reply:
Do we have scenario where `TraceState` once created would be modified? If not, it would be good to keep it static. This way, we can avoid creating empty instance with every span creation.

Reply:
I'm afraid I opened a can of worms here. As far as I understand, any `TraceState` object is read-only, any modifications returns a completely new object.

However, I think to make this interoperable with the changes here we would need those modifiers (`Get` and `Delete`) to return a `shared_ptr` instead of a `TraceState` instance on the stack: https://github.com/open-telemetry/opentelemetry-cpp/blob/main/api/include/opentelemetry/trace/trace_state.h#L228

Reply:
Agree, both `TraceState::Get` and `TraceState::Delete` should return shared_ptr objects.

Reply:
Ok, updated to return `shared_ptr` from `TraceState::Set`, `TraceState::Delete` and `TraceState::FromHeader`.